drivers/gpio/gpio_infineon_cat1.c

```diff
@@ -0,0 +1,332 @@
+}
+
+static int gpio_cat1_port_get_raw(const struct device *dev,
+				  uint32_t *value)
+{
+	GPIO_PRT_Type *const base = DEV_CFG(dev)->regs;
+
+	*value = GPIO_PRT_IN(base);
+
+	return 0;
+}
+
+static int gpio_cat1_port_set_masked_raw(const struct device *dev,
+					 uint32_t mask, uint32_t value)
+{
+	GPIO_PRT_Type *const base = DEV_CFG(dev)->regs;
+
+	GPIO_PRT_OUT(base) = (GPIO_PRT_OUT(base) & ~mask) | (mask & value);
+
+	return 0;
+}
+
+static int gpio_cat1_port_set_bits_raw(const struct device *dev,
+				       uint32_t mask)
+{
+	GPIO_PRT_Type *const base = DEV_CFG(dev)->regs;
+
+	GPIO_PRT_OUT_SET(base) = mask;
+
+	return 0;
+}
+
+static int gpio_cat1_port_clear_bits_raw(const struct device *dev,
+					 uint32_t mask)
+{
+	GPIO_PRT_Type *const base = DEV_CFG(dev)->regs;
+
+	GPIO_PRT_OUT_CLR(base) = mask;
+
+	return 0;
+}
+
+static int gpio_cat1_port_toggle_bits(const struct device *dev,
+				      uint32_t mask)
+{
+	GPIO_PRT_Type *const base = DEV_CFG(dev)->regs;
+
+	GPIO_PRT_OUT_INV(base) = mask;
+
+	return 0;
+}
+
+static uint32_t gpio_cat1_get_pending_int(const struct device *dev)
+{
+	const struct gpio_cat1_config *const cfg = DEV_CFG(dev);
+	GPIO_PRT_Type *const base = cfg->regs;
+
+	return GPIO_PRT_INTR_MASKED(base);
+}
+
+static void gpio_event_callback(void *callback_arg, cyhal_gpio_event_t event)
+{
+	ARG_UNUSED(event);
+	uint32_t port_num = CYHAL_GET_PORT((uint32_t) callback_arg);
+	uint32_t pin_num = CYHAL_GET_PIN((uint32_t) callback_arg);
+
+	static const struct device *const port_dev_obj[IOSS_GPIO_GPIO_PORT_NR] = {
+		GET_DEV_OBJ(gpio_prt0),  GET_DEV_OBJ(gpio_prt1),  GET_DEV_OBJ(gpio_prt2),
+		GET_DEV_OBJ(gpio_prt3),  GET_DEV_OBJ(gpio_prt4),  GET_DEV_OBJ(gpio_prt5),
+		GET_DEV_OBJ(gpio_prt6),  GET_DEV_OBJ(gpio_prt7),  GET_DEV_OBJ(gpio_prt8),
+		GET_DEV_OBJ(gpio_prt9),  GET_DEV_OBJ(gpio_prt10), GET_DEV_OBJ(gpio_prt11),
+		GET_DEV_OBJ(gpio_prt12), GET_DEV_OBJ(gpio_prt13), GET_DEV_OBJ(gpio_prt14)
+	};
+
+	const struct device *dev = port_dev_obj[port_num];
+
+	/* Goes through and fires callback from a callback list */
+	gpio_fire_callbacks(&DEV_DATA(dev)->callbacks, dev, 1 << pin_num);
+
+	/* NOTE: cyhal gpio handles cleaning of interrupts */
+}
+
+static int gpio_cat1_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					     enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_cat1_config *const cfg = DEV_CFG(dev);
+	cyhal_gpio_event_t event = CYHAL_GPIO_IRQ_NONE;
+
+	/* Level interrupts (GPIO_INT_MODE_LEVEL) is not supported */
+	if (mode == GPIO_INT_MODE_LEVEL) {
+		return -ENOTSUP;
+	}
+
+	switch (trig) {
+	case GPIO_INT_TRIG_LOW:
+		event = CYHAL_GPIO_IRQ_FALL;
+		break;
+
+	case GPIO_INT_TRIG_HIGH:
+		event = CYHAL_GPIO_IRQ_RISE;
+		break;
+
+	case GPIO_INT_TRIG_BOTH:
+		/* Trigger detection on pin rising or falling edge (GPIO_INT_TRIG_BOTH)
+		 * is not supported. Refer to SWINTEGRATION-696
+		 */
+		/* event = CYHAL_GPIO_IRQ_BOTH; */
+		return -ENOTSUP;
```

Comment:
you can just remove this case, default will return the same

Reply:
Keep this case for clear understanding that GPIO_INT_TRIG_BOTH doesn't support so far.